rfcs/0000-rda-claims-jsonb.md

```diff
@@ -0,0 +1,341 @@
+
+As such it creates code and design differences that could trip up new developers.
+However this is offset by:
+- The database schema is greatly simplified.
+- The likelihood of schema migrations being needed for future RDA API changes are greatly reduced.
+- The implementation is simple to understand and would work as-is with other entities in the future if desired.
+
+**Drawback 2:** Postgresql JSONB columns store the full JSON in binary form.
+
+This increases storage reqirements for the schema since the field names in the JSON are repeated in every record.
+The same benchmark that measured faster ingestion rates also measured increased storage requirements:
+- Storage per FISS claim increased from 1,460 bytes per claim to 1,958.
+- Storage per MCS claim increased from 1,036 bytes per claim to 1,822.
+The size measurement was not extremely precise as the size increased between updates.
+This variation was likely due to internal postgresql details.
+
+Although the overal storage in the database was increased somewhat with JSONB the overall performance would not be adversely affected.
+Since the JSONB column is not indexed, the size of the index for each column would be the same with JSONB as with the current schema.
+Overall I/O with the database should be lower with this schema since only one `SELECT`` is needed for each claim rather than one for the main table plus one for each detail table.
+
+**Drawback 3:** Without the relational structure we can't perform queries in the normal way.
+
+Standard SQL query syntax for fields within the JSON would not be available however postgresql provides a set of operators and functions for querying information within a `JSONB` column.
+This feature is documented here: https://www.postgresql.org/docs/12/functions-json.html
+
+The functions and operators are different than what a SQL user may be used to but they allow full access to values contained in the JSON.
+Postgresql also supports building indexes from properties of objects stored in a JSONB column although we plan to copy often used values into separate columns within the record so that BFD API queries don't have to depend on non-standard query syntax.
+
+
+### Proposed Solution: Notable Alternatives
+[Proposed Solution: Notable Alternatives]: #proposed-solution-notable-alternatives
+
+Since this proposal could be boiled down to using a NoSQL approach for FISS and MCS claims why not go all the way?
+A case could be made to move this data into Amazon's DynamoDB since it is built for this sort of design.
+While there could be advantages to that approach the proposed design has some practical advantages:
+- Using postgresql allows us to reap the benefits of a NoSQL approach without the need to add a whole new database technology to BFD.
+- RDS is already approved for storing PII so the changes can be implemented at any time without having to wait for review and approval of DynamoDB.
+
+
+## Prior Art
+[Prior Art]: #prior-art
```

Comment:
Not entirely sure this is true; while there may not be any prior art within BFD, document/NoSQL database are pretty pervasive these days.